### PR TITLE
Suggestion from upstream

### PR DESCRIPTION
**Pull request by cubap**
I've brought in the updates upstream that allow for TPEN() to handle auth without exposing anything. I handle things a little more simply and just read out the results on screen to prove it's working.

You can click on the title of the project to load the id automatically. If it is in the URL querystring it loads automatically, which is the expected default behavior.

Now you should have a start where you can work with the project data and move forward from there.

### CODE REVIEW

File: TPEN/events.mjs
Comment (cubap) [inline TPEN/events.mjs:1]: This is just the Class from upstream that ns the TPEN methods and constants.

File: TPEN/index.mjs
Comment (cubap) [inline TPEN/index.mjs:1]: This is just the Class from upstream that ns the TPEN methods and constants.

File: User/index.mjs
Comment (cubap) [inline User/index.mjs:1]: No changes, just bringing this up to date with `main`

File: _classes/TPEN.md
Comment (cubap) [inline _classes/TPEN.md:1]: No changes, just bringing this up to date with main

File: components/classroom/fetchData.mjs [hunk 1]
```diff
@@ -1,32 +1,33 @@
...
...cGVuIl0sImh0dHA6Ly9kdW5iYXIucmVydW0uaW8vYXBwX2ZsYWciOlsidHBlbiJdLCJodHRwOi8vcmVydW0uaW8vdXNlcl9yb2xlcyI6eyJyb2xlcyI6WyJkdW5iYXJfdXNlcl9wdWJsaWMiLCJnbG9zc2luZ191c2VyX3B1YmxpYyIsImxyZGFfdXNlcl9wdWJsaWMiLCJyZXJ1bV91c2VyX3B1YmxpYyIsInRwZW5fdXNlcl9wdWJsaWMiXX0sImh0dHA6Ly9kdW5iYXIucmVydW0uaW8vdXNlcl9yb2xlcyI6eyJyb2xlcyI6WyJkdW5iYXJfdXNlcl9wdWJsaWMiLCJnbG9zc2luZ191c2VyX3B1YmxpYyIsImxyZGFfdXNlcl9wdWJsaWMiLCJyZXJ1bV91c2VyX3B1YmxpYyIsInRwZW5fdXNlcl9wdWJsaWMiXX0sImlzcyI6Imh0dHBzOi8vY3ViYXAuYXV0aDAuY29tLyIsInN1YiI6ImF1dGgwfDY3MDk4NjczOTJlOTg1NDY5Yzc5OWY1YyIsImF1ZCI6WyJodHRwczovL2N1YmFwLmF1dGgwLmNvbS9hcGkvdjIvIiwiaHR0cHM6Ly9jdWJhcC5hdXRoMC5jb20vdXNlcmluZm8iXSwiaWF0IjoxNzI4ODUxNTcyLCJleHAiOjE3Mjg4NTg3NzIsInNjb3BlIjoib3BlbmlkIHByb2ZpbGUgZW1haWwgdXBkYXRlOmN1cnJlbnRfdXNlcl9tZXRhZGF0YSBvZmZsaW5lX2FjY2VzcyIsImF6cCI6ImJCdWdGTVdIVW8xT2huU1pNcFlVWHhpM1kxVUpJN0tsIn0.4jcJCJkc9bxo8VLaRhSbuB9fQxlOurhXE82tZ26TDU5Vp7QR7Y8qxBSm8f7hjvJe48gf4gvZD68KXCUabezAqlVe5MuUTLSnhmqfEHV5VJTmeIBmljCyrUowvrZWlzEMWV3EjdC-Cq4t8jVeFtr9DTOsHqx9k1sRaKIrkCCN1Ww8LFWqgW4ycL7cLg6lVARDKSjec-NB8kQwXPo-eaOPAcEgizUxx91teVufmrFX_9h8eR5HvdhEd9BWKdqGC9Ezjfqa1NLaJ_zDzsNe73xLgTH6BaO2EMPfvXw8ZWOhQdBGNQ_vGhCaU2WiOgeK7H24q9V2nhpGKWjCiWmGtdM8PA';
```
Comment (cubap) [inline components/classroom/fetchData.mjs:4]: No token in repo. Especially my token. At least it is expired.

File: components/classroom/fetchData.mjs [hunk 2]
```diff
@@ -1,32 +1,33 @@
...
-            headers: {
-                'Authorization': `Bearer ${token}`
-            }
-        });
+PROJECT_FORM.TPEN = new TPEN();
+TPEN.attachAuthentication(PROJECT_FORM);
 
-        if (!response.ok) {
-            throw new Error('Project not found or access denied');
+if (PROJECT_FORM.TPEN.activeProject?._id) {
```
Comment (cubap) [inline components/classroom/fetchData.mjs:8]: This is where the queryString will be loaded if it is included.

File: components/classroom/fetchData.mjs [hunk 3]
```diff
@@ -1,32 +1,33 @@
...
+    projectId.value = PROJECT_FORM.TPEN.activeProject._id;
+    const token = TPEN.getAuthorization();
```
Comment (cubap) [inline components/classroom/fetchData.mjs:10]: `false` if there isn't auth, but by now there definitely should be.

File: components/classroom/fetchData.mjs [hunk 4]
```diff
@@ -1,32 +1,33 @@
...
+    // until the Project class is better defined, we'll just fetch the project data directly
+    const project = await fetch(`${PROJECT_FORM.TPEN.servicesURL}/project/${PROJECT_FORM.TPEN.activeProject._id}`, {
+        method: 'GET',
```
Comment (cubap) [inline components/classroom/fetchData.mjs:13]: this fetch() isn't different from what you were doing and should eventually be obscured behind an incoming Project class for Interfaces.

File: components/classroom/fetchData.mjs [hunk 5]
```diff
@@ -1,32 +1,33 @@
...
-    }
+    })
+        .then(res => res.ok ? res.json() : Promise.reject(res.status))
+        .then(data => msg.innerHTML = `<pre>${JSON.stringify(data, null, 2)}</pre>`)
+        .catch(err => msg.innerHTML = `<pre>${JSON.stringify(err, null, 2)}</pre>`);
+}
+
+PROJECT_FORM.addEventListener('submit', async function (event) {
+    const projectID = projectId.value;
+    location.href = '?projectID=' + projectID;  // Redirect to the same page with the projectID in the query string
```
Comment (cubap) [inline components/classroom/fetchData.mjs:25]: This just loads the page with the queryString, so the processing happens naturally.

File: components/classroom/index.html [hunk 6]
```diff
@@ -1,50 +1,67 @@
 <!DOCTYPE html>
 <html lang="en">
+
 <head>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <link rel="stylesheet" type="text/css" href="./styles.css" />
     <title>TPEN Classroom Interface</title>
+    <script src="../projects-list/index.mjs" type="module"></script>
```
Comment (cubap) [inline components/classroom/index.html:9]: This component is new and just the list of projects titles

File: components/classroom/index.html [hunk 7]
```diff
@@ -1,50 +1,67 @@
...
+    <script type="module" src="./fetchData.mjs"></script>
+    <style>
```
Comment (cubap) [inline components/classroom/index.html:11]: local styles just to make the page readable without polluting your actual real CSS.

File: components/classroom/index.html [hunk 8]
```diff
@@ -1,50 +1,67 @@
...
+    <div class="page-container">
+        <h1>Welcome!</h1>
+        <div class="input-container">
+            <div class="label-container">
+                <form id="projectForm" class="input-form">
+                    <label for="projectId">Project ID:</label>
+                    <input id="projectId" maxlength="24" name="projectId" type="text" required>
+                    <button class="submit-button" type="submit">Submit</button>
+                </form>
+                <tpen-projects-list id="clickList"></tpen-projects-list>
```
Comment (cubap) [inline components/classroom/index.html:52]: imported component from above

File: components/classroom/index.html [hunk 9]
```diff
@@ -1,50 +1,67 @@
...
-        <footer>
-            <p>TPEN Classroom Interface</p>
-        </footer>
+    <footer>
+        <p>TPEN Classroom Interface</p>
+    </footer>
+</body>
 
-        <script src="./menutoggle.js"></script>
-        <script type="module" src="./fetchData.mjs"></script>
```
Comment (cubap) [inline components/classroom/index.html:48]: This is in the wrong place. The type=module already defers its loading, so there is no good reason to drop scripts down here.

File: components/default-transcribe/index.html
Comment (cubap) [inline components/default-transcribe/index.html:1]: No changes, just bringing this up to date with main

File: components/default-transcribe/index.js
Comment (cubap) [inline components/default-transcribe/index.js:1]: No changes, just bringing this up to date with main

File: components/default-transcribe/main.js
Comment (cubap) [inline components/default-transcribe/main.js:1]: No changes, just bringing this up to date with main

File: components/iiif-tools/index.mjs
Comment (cubap) [inline components/iiif-tools/index.mjs:1]: No changes, just bringing this up to date with main

File: components/projects-list/index.html
Comment (cubap) [inline components/projects-list/index.html:1]: No changes, just bringing this up to date with main

File: components/projects-list/index.mjs
Comment (cubap) [inline components/projects-list/index.mjs:1]: No changes, just bringing this up to date with main

File: components/public-login/index.js
Comment (cubap) [inline components/public-login/index.js:1]: No changes, just bringing this up to date with main

File: components/user-profile/index.html
Comment (cubap) [inline components/user-profile/index.html:1]: No changes, just bringing this up to date with main

File: components/user-profile/index.mjs
Comment (cubap) [inline components/user-profile/index.mjs:1]: No changes, just bringing this up to date with main

File: favicon.ico
Comment (cubap) [inline favicon.ico:1]: brand new. You're going to love it.

File: index.html
Comment (cubap) [inline index.html:1]: No changes, just bringing this up to date with main

File: manage/projectsHighlight.mjs
Comment (cubap) [inline manage/projectsHighlight.mjs:1]: Standard daily dosing is 800 IU daily and 1200 mg calcium daily

File: projects/index.mjs
Comment (cubap) [inline projects/index.mjs:1]: No changes, just bringing this up to date with main

**Review by MutyaSowmya123 — APPROVED** [review]
I see all the changes are required for better working of the interface.!

**Pull request merged**